File: frontend/src/api/catalog.ts
import api from "./client";

export interface Product {
    id: number;
    name_es: string;
    name_en?: string;
    is_active: boolean;
}

export interface Unit {
    id: number;
    name: string;
    abbreviation?: string;
    is_active: boolean;
}

export const fetchProducts = async () => {
    const { data } = await api.get<Product[]>("/catalog/products");
    return data;
};

export const createProduct = async (product: { name_es: string; name_en?: string }) => {
    const { data } = await api.post<Product>("/catalog/products", product);
    return data;
};

export const deleteProduct = async (id: number) => {
    await api.delete(`/catalog/products/${id}`);
};

export const fetchUnits = async () => {
    const { data } = await api.get<Unit[]>("/catalog/units");
    return data;
};

export const createUnit = async (unit: { name: string; abbreviation?: string }) => {
    const { data } = await api.post<Unit>("/catalog/units", unit);
    return data;
};

export const deleteUnit = async (id: number) => {
    await api.delete(`/catalog/units/${id}`);
};

File: backend/app/models/catalog.py
from sqlalchemy import Column, Integer, String, Boolean, Text
from app.models.base import Base


class Product(Base):
    __tablename__ = "products"

    id = Column(Integer, primary_key=True, index=True)
    name_es = Column(String, unique=True, index=True, nullable=False)
    name_en = Column(String, nullable=True)
    is_active = Column(Boolean, default=True)


class Unit(Base):
    __tablename__ = "units"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, unique=True, nullable=False)
    abbreviation = Column(String, nullable=True)
    is_active = Column(Boolean, default=True)


class SavedStop(Base):
    """
    Catálogo de paradas/tiradas guardadas para autocompletado.
    Se van creando conforme se usan en cotizaciones.
    """
    __tablename__ = "saved_stops"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(255), unique=True, index=True, nullable=False)  # Nombre identificador de la parada
    address = Column(Text, nullable=True)  # Dirección completa
    city = Column(String(100), nullable=True)  # Ciudad
    state = Column(String(100), nullable=True)  # Estado/Provincia
    country = Column(String(50), nullable=True, default="USA")  # País
    default_contact = Column(String(255), nullable=True)  # Contacto por defecto
    default_phone = Column(String(50), nullable=True)  # Teléfono por defecto
    default_schedule = Column(String(100), nullable=True)  # Horario típico ej: "8:00 AM - 5:00 PM"
    notes = Column(Text, nullable=True)  # Notas adicionales
    is_active = Column(Boolean, default=True)
